docs/workforce-human-oidc-auth.md

```diff
@@ -0,0 +1,267 @@
+Device Auth only if
+
+- The former is unavailable under specific circumstances, which are described in the Auth Code Flow section; and
+- The user has explicitly indicated that Device Auth is enabled for this connection attempt.
+
+In both of these cases, the general chain of events is as follows:
+
+1. The user passes a connection string to the client application and indicates (possibly through said connection string)
+    that they desire to use `MONGODB-OIDC` as their authentication mechanism.
+2. The application connects to the MongoDB endpoint and asks it for information about the Identity Provider (IdP)[^1] it
+    is supposed to authenticate with. If there was a username in the connection string, the application sends that as
+    well.
+3. The MongoDB endpoint responds with information about the IdP, in the form of an `issuer` URL identifying the IdP and
+    other metadata about it.
+4. The application looks up the public metadata document for the IdP, which includes information about user-facing URL
+    endpoints.
+5. The application composes a URL based on those endpoints and presents it to the user in some form (the details depend
+    on the exact flow chosen).
+6. The user visits that URL, which can result in any number of steps to authenticate the user.
+7. The application receives a token from the IdP (the details again depend on the exact flow chosen).
+8. The application presents this token to the MongoDB endpoint.
+
+Since in these steps the client application performs actions depending on information supplied by external sources, this
+document specifies mechanisms that help ensure that these actions are carried out securely. MongoDB clients are “public
+clients” in the sense of the OIDC specs (i.e. they do not possess a shared secret with the Identity Provider), and must
+behave accordingly.
+
+Since these steps require the involvement of a human being, the application should be set up to gracefully handle
+timeouts and provide the user with a way to abort the connection attempt.
+
+# General requirements
+
+An application implementing workforce OIDC needs to generally comply with OpenID and OAuth 2.0 standards.
+
+In particular:
+
+- Any HTTP calls made to non-local servers MUST be made using HTTPS.
+- IdP metadata discovery MUST be implemented as described in [RFC8414](https://datatracker.ietf.org/doc/html/rfc8414).
+
+Using a well-tested and standards-compliant third-party library for core OIDC logic for the respective ecosystem is
+highly recommended. If this is not possible, implementers need to pay special attention to the specifications referenced
+in this document.
+
+## Endpoint restrictions
+
+Currently, users who connect to a host other than localhost or an Atlas hostname need to explicitly opt-in into being
+able to do so by setting the `ALLOWED_HOSTS` flag (specified in the drivers auth spec). In the future, MongoDB is hoping
```

Comment:
Awkward wording:
>explicitly opt-in into being able to do so by setting the `ALLOWED_HOST` flag

Maybe something like this instead:
>explicitly specify this host via `ALLOWED_HOSTS`

Reply:
Thanks, absolutely right – done in b66f9609e7e608953241aa43a1711865f1f6c1ab 👍